src/function/combinatorics/stirlingS2.js

```diff
@@ -62,18 +73,32 @@ export const createStirlingS2 = /* #__PURE__ */ factory(name, dependencies, (
         throw new TypeError('k must be less than or equal to n in function stirlingS2')
       }
 
-      // 1/k! Sum(i=0 -> k) [(-1)^(k-i)*C(k,j)* i^n]
-      const kFactorial = factorial(k)
-      let result = 0
-      for (let i = 0; i <= k; i++) {
-        const negativeOne = pow(-1, subtract(k, i))
-        const kChooseI = combinations(k, i)
-        const iPower = pow(i, n)
-
-        result = addScalar(result, multiplyScalar(multiplyScalar(kChooseI, iPower), negativeOne))
+      const big = !(isNumber(n) && isNumber(k))
+      const cache = big ? bigCache : smallCache
+      const make = big ? bignumber : number
+      const nn = number(n)
+      const nk = number(k)
+      /* See if we already have the value: */
+      if (cache[nn] && cache[nn].length > nk) {
```

Comment:
I guess the caching makes the implementation hugely faster. How large can the cache grow in practice? Do we have to be careful there?

Reply:
Yes using the recurrence is probably impractical without caching. Because Stirling grows so fast, the cache can't get too huge: with numbers you can practically compute to S2(23,23), so about 600 number cells in all, and with bignumbers I didn't push it but I am guessing you can reasonably go up to something like S2(100,100) so 10,000 bignumber cells. These don't seem onerous to me...

Reply:
These amounts should indeed be no problem. Ok nice, then I think it's ready to merge.